dll/win32/shell32/wine/shpolicy.c

```diff
@@ -39,769 +39,227 @@
 #include <windef.h>
 #include <winbase.h>
 #include <shlobj.h>
+#include <initguid.h>
+#include <shlwapi_undoc.h>
 #include <wine/debug.h>
 
 #include "shell32_main.h"
 
 WINE_DEFAULT_DEBUG_CHANNEL(shell);
 
-#define SHELL_NO_POLICY 0xffffffff
+DEFINE_GUID(GUID_Restrictions, 0xA48F1A32, 0xA340, 0x11D1, 0xBC, 0x6B, 0x00, 0xA0, 0xC9, 0x03, 0x12, 0xE1);
 
-typedef struct tagPOLICYDAT
+static const POLICYDATA s_PolicyTable[] =
 {
-  DWORD policy;          /* policy value passed to SHRestricted */
-  LPCSTR appstr;         /* application str such as "Explorer" */
-  LPCSTR keystr;         /* name of the actual registry key / policy */
-  DWORD cache;           /* cached value or 0xffffffff for invalid */
-} POLICYDATA, *LPPOLICYDATA;
-
-/* application strings */
+    /* { policy, appstr, keystr }, */
```

Comment:
please keep the original formatting of the table, it makes it much easier to read.
(Also the original one documented the versioning and any policy name variation between shell32 versions, while all this information is lost in your PR).

Reply:
Win2k3 `shell32` has no such data.

Reply:
This table should be compatible.
Wine's table is not compatible.

Reply:
And in what sense wine's table is not compatible? (Note that I'm not talking about the fourth field in each entry: "cache", that you've removed)

Reply:
2k3 POLICYDATA has no cache entry.
if you added this cache entry, then you can't call shlwapi SHRestrictionLookup function safely. Because incompatible.

Reply:
Ah no, I didn't add the cache entry, it was already there, see the last few commits in master.
You could take the table from master and just remove that cache entry.

Reply:
I think it should be a list of functional macros for code flexibility and machine readability.

Reply:
> I think it should be a list of functional macros for code flexibility and machine readability.

I don't know. Is there a lot of data that repeats between different entries? If so, maybe, but otherwise I'm not sure that would help.
Unless, the contents of this table have somehow (??) to be duplicated in two separate modules, where the entries in these modules were similar except for some extra stuff. Then using macros this could help (the two modules have a same-named macro with same parameters, but with different definitions.... I think you've done that already for some IME code.)

Reply:
I wasn't refering to that kind of duplication, but instead about the pros vs. cons of using a macro just for doing what's already done via table entry.

Reply:
Done.